src/test/test_pidfile.sh

```diff
@@ -0,0 +1,111 @@
+#!/bin/bash -e
```

Comment:
i  test if we remove -e, then run_mon $dir a "fail" will never fail even though mon actually fails.
at this situation, we have to add return 1 after every run_mon or run_osd.  so keeping -e may look morebeautiful then seeing return 1 everwhere.